### PR TITLE
Dk3750

### PR DESCRIPTION
**Pull request by sondrele**
Depends on #25 to be merged before this one.
Makes it possible to compile to either the STK3700 or the DK3750 by passing `KIT` to make (defaults to stk37000).
E.g: `make PROJ_NAME=usart KIT=dk3750`. This will compile kit-specific C-dependencies, as well as kit-specific drivers/bsp-bindings .

This branch has also refactored the `stk` module to `emlib::kits::stk3700`, for a more logical structure in order to support more kits. This change requires that either this branch is merged before #24 , or that #24 applies the changes and moves its `stk::bsp` module to `kits::stk3700::bsp` after this branch is merged into master.

### CODE REVIEW

**Comment by havardh** [comment]
Rebase this on master and fix module naming.

File: build/emlib.rs [hunk 1]
```diff
@@ -57,14 +57,33 @@ fn base_config(config: &mut Config) -> &mut Config {
...
+    println!("Configuring STK3700");
+    common_config(config)
+        .include("efm32-common/kits/EFM32GG_STK3700/config")
+
+        .file("efm32-common/kits/common/bsp/bsp_stk.c")
+        .file("efm32-common/kits/common/bsp/bsp_bcc.c")
+}
 
+#[allow(dead_code)]
+fn dk3750_base_config(config: &mut Config) -> &mut Config {
```
Comment (havardh) [inline build/emlib.rs]: Move the `{dk3750, sdk3700}_base_config` into their own files.

Comment (havardh) [inline build/emlib.rs]: Rename to `kit_config`

File: build/emlib.rs [hunk 2]
```diff
@@ -10,6 +10,16 @@ use std::fs::File;
 use std::io::Error;
 use std::io::prelude::*;
 
+#[cfg(feature = "dk3750")]
+use dk3750::kit_config;
+
+#[cfg(feature = "stk3700")]
+use dk3750::kit_config;
```
Comment (havardh) [inline build/emlib.rs]: stk3700 ?

**Comment by havardh** [comment]
I get this when building.

```
In file included from /vagrant/efm32-common/kits/common/bsp/bsp_dk_3201.c:24:0:
/vagrant/efm32-common/kits/common/bsp/bsp.h:22:23: fatal error: bspconfig.h: No such file or directory
 #include "bspconfig.h"
```

**Comment by sondrele** [comment]
Is `#include "bspconfig.h"` still a problem with vagrant?

**Pull request closed without merging**